### PR TITLE
feature(footer): 新增td_footer组件

### PR DESCRIPTION
**Pull request by chendingya**
### 🤔 这个 PR 的性质是？

- [ ] 日常 bug 修复
- [x] 新特性提交
- [ ] 文档改进
- [ ] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] CI/CD 改进
- [ ] 重构
- [ ] 代码风格优化
- [ ] 测试用例
- [ ] 分支合并
- [ ] 其他

### 🔗 相关 Issue
https://github.com/Tencent/tdesign-flutter/issues/213

### 💡 需求背景和解决方案
按照设计稿完成Footer组件开发，设计稿地址：https://codesign.qq.com/s/332865

image-20240707212959006

### 📝 更新日志

- feature(td_footer): 新增td_footer组件

- [ ] 本条 PR 不需要纳入 Changelog

### ☑️ 请求合并前的自查清单

⚠️ 请自检并全部**勾选全部选项**。⚠️

- [ ] 文档已补充或无须补充
- [x] 代码演示已提供或无须提供
- [ ] Changelog 已提供或无须提供

### CODE REVIEW

File: tdesign-component/lib/src/components/footer/td_footer.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import '../../../tdesign_flutter.dart';
+
+
+class TDFooter extends StatefulWidget {
+  const TDFooter({
+    Key? key,
+    this.logo,
+    this.text = 'Copyright © 2019-2023 TDesign.All Rights Reserved.',
```
Comment (Luozf12345) [inline tdesign-component/lib/src/components/footer/td_footer.dart:10]: 这里请不要设置有特殊含义的默认值，可以是空字符串或者空对象

File: tdesign-component/lib/src/components/footer/td_footer.dart [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        Padding(
+          padding: const EdgeInsets.only(top: 4, bottom: 4),
+          child: Row(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: widget.links.map((link) {
+              return Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 4),
+                child: TDLink(
```
Comment (Luozf12345) [inline tdesign-component/lib/src/components/footer/td_footer.dart:89]: 修复了TDLink的下划线不展示问题，请同步一下主仓库的develop分支

File: tdesign-component/lib/src/components/footer/td_footer.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          else
+            _renderText(),
+        ],
+      ),
+    );
+  }
+
+  Widget _renderLogo() {
+    return Container(
+        color: TDTheme.of(context).whiteColor1,
```
Comment (Luozf12345) [inline tdesign-component/lib/src/components/footer/td_footer.dart:56]: 无需默认背景色。如果添加背景色，需要允许用户自定义，否则放在不是白色背景的界面，显示会有问题。

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/tdesign-flutter?pullRequest=224) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/tdesign-flutter?pullRequest=224) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: Luozf12345<br/>:white_check_mark: chendingya<br/>:x: aaronmhl<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/tdesign-flutter?pullRequest=224) it.</sub>

File: tdesign-component/example/lib/page/td_footer_page.dart [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  const TDFooterPage({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _TDFooterPageState();
+}
+
+class _TDFooterPageState extends State<TDFooterPage> {
+  @override
+  Widget build(BuildContext context) {
+    return ExamplePage(
```
Comment (Luozf12345) [inline tdesign-component/example/lib/page/td_footer_page.dart:17]: 如果能吧backgroundColor设为白色会更好

**Review by Luozf12345 — APPROVED** [review]

**Review by Luozf12345 — APPROVED** [review]

**Pull request merged**